fuse_models/include/fuse_models/unicycle_2d.h

```diff
@@ -189,6 +189,9 @@ class Unicycle2D : public fuse_core::AsyncMotionModel
   fuse_core::Matrix8d process_noise_covariance_;   //!< Process noise covariance matrix
   bool scale_process_noise_{ false };              //!< Whether to scale the process noise covariance pose by the norm
                                                    //!< of the current state twist
+  bool rotate_process_noise_covariance_to_state_orientation_{ false };  //!< Whether to rotate the process noise
```

Comment:
As I said before, I wonder if this should always be done. I sounds like it's probably the right thing to do. In that case, I think we should always do it, and we wouldn't need a ROS param at all. :thinking:

Reply:
I also believe that this should always be done and would enable it for all our robots. That's why I now also removed the ROS parameter with the latest commit.
The only reason why I created a ROS parameter before and set it to false by default was that existing applications using fuse are not changed and remain backwards compatibility. If you agree that this is no issue, then I would be happy to have it always enabled. Otherwise I will just revert the last commit and I'm fine with a ROS parameter too.

Reply:
If the math is wrong, then I would consider it a bug. I see no need for an option to retain the wrong math either.

Reply:
How can we confirm it was a bug?

Is it something we can clearly see from the math?

@fabianhirmann Would you be able to provide a unit test that only passes with your changes but didn't before? :thinking:

Reply:
I just pushed a unit test for it that does not pass before my changes but only with my changes. There I also wrote an explanation why I expect that the final process noise covariance should be rotated. Summarized my perspective is that the process noise covariance is defined in the **robot** frame so when the position is defined in the **world** frame, we need to rotate the process noise covariance to the world frame (using the yaw angle that defines the orientation relation between the robot frame and world frame). 
An example, imagine we are defining the process noise covariance with `cov_robot_xx = 2.0` and `cov_robot_yy = 1.0` (assuming that the process noise covariance is in the **robot** frame there is then less process noise covariance in the lateral direction than the longitudinal direction). Then, when having `yaw angle = 90°`, I would expect that the final process noise covariance applied to the position in the **world** frame would have `cov_world_xx = 1.0` and `cov_world_yy = 2.0` because the orientation of the robot is upwards and the covariance must then get 90° rotated.